internal/lm/nvml.go

```diff
@@ -254,3 +353,20 @@ func getDeviceClasses(devices []resource.Device) ([]uint32, error) {
 	}
 	return classes, nil
 }
+
+// generateUUID generates a UUID-like string based on an MD5 hash of the seed and random bytes.
+func generateUUID(seed string) string {
```

Comment:
Question: Should this really be random? Given the same input, should we not expect the UUID to be the same?

Does using https://pkg.go.dev/hash/maphash@go1.23.1 not server our purposes here?

Reply:
It's not random -- it just uses the `rand` package sourced with a specific seed to guarantee it always gives us back the same value.

Reply:
There's likely no reason we couldn't use the crypto package for the same purpose though -- I just used `rand` when I first wrote this because it was the first one I found that did what I needed.